### PR TITLE
[FLINK-16074][docs-zh] Translate the Overview page for State & Fault Tolerance into Chinese

### PR DESCRIPTION
**Pull request by klion26**
## What is the purpose of the change

Translate the Overview page for State & Fault Tolerance into Chinese

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable)

CC  @carp84 @wuchong

### CODE REVIEW

**Comment by klion26** [comment]
The translation first aligned with the [English version](https://ci.apache.org/projects/flink/flink-docs-master/dev/stream/state/), then translates the newly version

File: docs/dev/stream/state/index.zh.md [hunk 1]
```diff
@@ -25,34 +25,17 @@ specific language governing permissions and limitations
...
-
-Flink needs to be aware of the state in order to make state fault tolerant using [checkpoints](checkpointing.html) and to allow [savepoints]({{ site.baseurl }}/ops/state/savepoints.html) of streaming applications.
-
-Knowledge about the state also allows for rescaling Flink applications, meaning that Flink takes care of redistributing state across parallel instances.
-
-The [queryable state](queryable_state.html) feature of Flink allows you to access state from outside of Flink during runtime.
-
-When working with state, it might also be useful to read about [Flink's state backends]({{ site.baseurl }}/ops/state/state_backends.html). Flink provides different state backends that specify how and where state is stored. State can be located on Java's heap or off-heap. Depending on your state backend, Flink can also *manage* the state for the application, meaning Flink deals with the memory management (possibly spilling to disk if necessary) to allow applications to hold very large state. State backends can be configured without changing your application logic.
-
+你将在本节中了解到 Flink 提供的用于编写有状态程序的 API，想了解更多有状态流处理的概念，请查考[有状态的流处理]({% link concepts/stateful-stream-processing.zh.md %})
```
Comment (Zakelly) [inline docs/dev/stream/state/index.zh.md:28]: I think maybe this is typo? The "查考" should be "参考" or "查看"

File: docs/dev/stream/state/index.zh.md [hunk 2]
```diff
@@ -25,34 +25,17 @@ specific language governing permissions and limitations
...
-* [Working with State](state.html): Shows how to use state in a Flink application and explains the different kinds of state.
-* [The Broadcast State Pattern](broadcast_state.html): Explains how to connect a broadcast stream with a non-broadcast stream and use state to exchange information between them. 
-* [Checkpointing](checkpointing.html): Describes how to enable and configure checkpointing for fault tolerance.
-* [Queryable State](queryable_state.html): Explains how to access state from outside of Flink during runtime.
-* [State Schema Evolution](schema_evolution.html): Shows how schema of state types can be evolved.
-* [Custom Serialization for Managed State](custom_serialization.html): Discusses how to implement custom serializers, especially for schema evolution.
+* [Working with State](state.html): 描述了如何在 Flink 应用程序中使用状态，以及不同类型的状态。
+* [The Broadcast State 模式](broadcast_state.html): 描述了如何将广播流和非广播流进行连接从而交换数据。
+* [Checkpointing](checkpointing.html): 介绍了如何开启和配置 checkpoint，以实现状态容错。
+* [可查询状态](queryable_state.html): 介绍了如何从外围访问 Flink 的状态。
```
Comment (carp84) [inline docs/dev/stream/state/index.zh.md:37]: 这个我们保留英文（Queryable State）吧

**Comment by klion26** [comment]
@carp84 thanks for the review, addressed the comment, please take another look when you're free.

**Comment by klion26** [comment]
Rebased the newly master

**Comment by klion26** [comment]
@carp84 thanks for the review and merging.

**Pull request closed without merging**